src/titan.rs

```diff
@@ -135,6 +138,16 @@ impl TitanBlobIndex {
         }
         Ok(index)
     }
+
+    pub fn encode_to(index: &TitanBlobIndex) -> Vec<u8> {
```

Comment:
Why not `pub fn encode(&self) -> Vec<u8>`?

Reply:
keep consistent with `decode_from`. I think both of these two function declaration are the same.

Reply:
I prefer index.encode_to(&mut vec), Self::decode_from(&[u8]) -> Self.

Reply:
I'm not sure what's the difference?

Reply:
When you have a `TitanBlobIndex`, why not use `index.encode()` instead of `TitanBlobIndex::encode_to(index)`? In addition, it does not encode to the `index`, but encodes the `index` to a buffer.